cache-config/t3c-tail/t3c-tail.go

```diff
@@ -0,0 +1,131 @@
+package main
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"regexp"
+	"time"
+
+	"github.com/apache/trafficcontrol/cache-config/t3cutil"
+	"github.com/apache/trafficcontrol/lib/go-log"
+
+	"github.com/nxadm/tail"
+	"github.com/pborman/getopt/v2"
+)
+
+const AppName = "t3c-tail"
+
+// Version is the application version.
+// This is overwritten by the build with the current project version.
+var Version = "0.4"
+
+// GitRevision is the git revision the application was built from.
+// This is overwritten by the build with the current project version.
+var GitRevision = "nogit"
+
+//default time out is 15 seconds, if not included in json input.
+var timeOutSeconds = 15
+
+func main() {
+	version := getopt.BoolLong("version", 'V', "Print version information and exit.")
+	help := getopt.BoolLong("help", 'h', "Print usage information and exit")
+	getopt.Parse()
+
+	log.Init(os.Stderr, os.Stderr, os.Stderr, os.Stderr, os.Stderr)
+
+	if *help {
+		fmt.Println(usageStr())
+		os.Exit(0)
+	} else if *version {
+		fmt.Println(t3cutil.VersionStr(AppName, Version, GitRevision))
+		os.Exit(0)
+	}
+
+	tailCfg := &t3cutil.TailCfg{}
+	if err := json.NewDecoder(os.Stdin).Decode(tailCfg); err != nil {
```

Comment:
Would you object to making these settings (`file`, `match`, `end-match`, `timeout`) flags instead of `stdin`? 
I think flags are more common and expected for Linux/POSIX apps, when they're static values not at risk of being arbitrarily long and overflowing Linux command buffers.

Reply:
Also, mind making the timeout in milliseconds, e.g. `--timeout-ms`? Users might want sub-second timeouts

Reply:
Changed these to flags. I was on the fence about flags vs stdin, and almost changed it before opening the PR.